### PR TITLE
Shared physics wsm6

### PR DESCRIPTION
**Pull request by weiwangncar**
TYPE: enhancement, new feature

KEYWORDS: shared physics, WSM6

SOURCE: internal

DESCRIPTION OF CHANGES:
Add WSM6 microphysics scheme from the shared physics directory, which is ccpp-compliant.

Solution:
The CCPP-compliant WSM6 MP code will be residing in phys/physics_mmm/ directory. Currently it is a copy of the code from the future MMM-physics repo.

LIST OF MODIFIED FILES: 
M       arch/postamble
M       arch/preamble
M       clean
M       main/depend.common
M       phys/Makefile
M       phys/module_microphysics_driver.F
M       phys/module_mp_wsm6.F
M       phys/module_physics_init.F
A       phys/physics_mmm/module_libmassv.F90
A       phys/physics_mmm/mp_radar.F90
A       phys/physics_mmm/mp_wsm6.F90
A       phys/physics_mmm/mp_wsm6_effectRad.F90
M       share/module_model_constants.F

TESTS CONDUCTED: 
1. The runs before and after this change produce bit-for-bit results when compiled with -d option.
2. Are the Jenkins tests all passing?

RELEASE NOTE: Add WSM6 microphysics scheme from the future shared MMM-physics repository.

### CODE REVIEW

**Comment by weiwangncar** [comment]
The regression test results:

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           57
    Number of Simulations  : 158           150        0
    Number of Comparisons  : 95           86        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

File: phys/module_physics_init.F [hunk 1]
```diff
@@ -4380,7 +4380,7 @@ SUBROUTINE mp_init(RAINNC,SNOWNC,GRAUPELNC,config_flags,restart,warm_rain,
 !------------------------------------------------------------------
    USE module_mp_wsm3
    USE module_mp_wsm5
-   USE module_mp_wsm6
+   USE mp_wsm6
```
Comment (islas) [inline phys/module_physics_init.F:4383]: I think the corresponding depend in depend.common needs to change to point to this module now

File: phys/module_physics_init.F [hunk 2]
```diff
@@ -4479,7 +4483,7 @@ SUBROUTINE mp_init(RAINNC,SNOWNC,GRAUPELNC,config_flags,restart,warm_rain,
      CASE (WSM5SCHEME)
           CALL wsm5init(rhoair0,rhowater,rhosnow,cliq,cpv, allowed_to_read )
      CASE (WSM6SCHEME)
-          CALL wsm6init(rhoair0,rhowater,rhosnow,cliq,cpv, config_flags%hail_opt,allowed_to_read )
+          CALL mp_wsm6_init(rhoair0,rhowater,rhosnow,cliq,cpv,config_flags%hail_opt,errmsg,errflg)
```
[the remark this answers is not in the document]
Comment (weiwangncar) [reply phys/module_physics_init.F:4486]: @dudhia Not sure what you mean....

File: phys/module_physics_init.F (hunk 1 above)
Comment (weiwangncar) [reply phys/module_physics_init.F:4383]: @islas This should be fixed.

File: phys/module_physics_init.F (hunk 2 above)
[the remark this answers is not in the document]
Comment (dudhia) [reply phys/module_physics_init.F:4486]: It must just be the way it shows up on my diffs then where CALL is on its own line

**Review by dudhia — APPROVED** [review]

**Review by smileMchen — APPROVED** [review]

**Comment by dudhia** [comment]
Is it true that the effectRad has been reorganized to include the *_bg
arrays as part of this?

**Comment by weiwangncar** [comment]
@dudhia I believe so: I see code 
'''
 real(kind=kind_phys),intent(in):: re_qc_bg,re_qi_bg,re_qs_bg
'''

**Review by dudhia — APPROVED** [review]
more changes than expected to effectRad

**Pull request merged**